config/initializers/sidekiq.rb

```diff
@@ -8,38 +8,40 @@
 
 Sidekiq::Client.try(:reliable_push!) unless Rails.env.test?
 
-Sidekiq.configure_server do |config|
-  config.try(:reliable!)
+Rails.application.config.to_prepare do
+  Sidekiq.configure_server do |config|
+    config.try(:reliable!)
 
-  config.redis = ThreeScale::RedisConfig.new(System::Application.config.sidekiq).config
-  config.error_handlers << System::ErrorReporting.method(:report_error)
+    config.redis = ThreeScale::RedisConfig.new(System::Application.config.sidekiq).config
+    config.error_handlers << System::ErrorReporting.method(:report_error)
 
-  config.server_middleware do |chain|
-    chain.add ThreeScale::Analytics::SidekiqMiddleware
-    chain.add ThreeScale::SidekiqRetrySupport::Middleware
-  end
+    config.server_middleware do |chain|
+      chain.add ThreeScale::Analytics::SidekiqMiddleware
+      chain.add ThreeScale::SidekiqRetrySupport::Middleware
+    end
 
-  faraday = ThreeScale::Core.faraday
-  faraday.options.timeout = 30
-  faraday.options.open_timeout = 10
+    faraday = ThreeScale::Core.faraday
+    faraday.options.timeout = 30
+    faraday.options.open_timeout = 10
 
-  schedule_file = Rails.root.join('config', 'sidekiq_schedule.yml')
+    schedule_file = Rails.root.join('config', 'sidekiq_schedule.yml')
 
-  if File.exist?(schedule_file) && Sidekiq.server?
-    Sidekiq::Cron::Job.load_from_hash! YAML.load_file(schedule_file)
+    if File.exist?(schedule_file) && Sidekiq.server?
+      Sidekiq::Cron::Job.load_from_hash! YAML.load_file(schedule_file)
+    end
+    # This will start a webrick server in another thread
```

Comment:
can only `Sidekiq.configure_server` be placed inside the `to_prepare`? I feel a little uneasy to see `    Yabeda::Prometheus::Exporter.start_metrics_server!` in the reloadable block.